pkg/api/v2/reporter/reporter.go

```diff
@@ -30,9 +30,63 @@ func (e ResourceReports) Accept(res ...resources.InputResource) ResourceReports
 	return e
 }
 
+// Merge merges the given resourceReports into this resourceReports.
+// Any resources which appear in both resourceReports will
+// have their warnings and errors merged.
+// Errors appearing in both reports, as determined by the error strings,
+// will not be duplicated in the resulting merged report.
 func (e ResourceReports) Merge(resErrs ResourceReports) {
 	for k, v := range resErrs {
-		e[k] = v
+		if firstReport, exists := e[k]; exists {
+			// report already exists for this resource,
+			// merge new report into existing report:
+			secondReport := v
+
+			// Merge warnings lists
+			allWarnings := make(map[string]bool)
+			for _, warning := range firstReport.Warnings {
+				allWarnings[warning] = true
+			}
+			for _, warning := range secondReport.Warnings {
+				if _, found := allWarnings[warning]; !found {
+					firstReport.Warnings = append(firstReport.Warnings, warning)
+				}
+			}
+
+			if firstReport.Errors == nil {
+				// Only 2nd has errs
+				firstReport.Errors = secondReport.Errors
+				e[k] = firstReport
+				continue
+			} else if secondReport.Errors == nil {
+				// Only 1st has errs
```

Comment:
don't we still need `e[k] = firstReport` here, in the event we had warnings but no errors in the second report?

Reply:
Great catch, added this + added a test for this case!